### PR TITLE
Adds Meson

### PR DESCRIPTION
**Pull request by dirmeier**
Meson is a modern build system for C, C++, D, Fortran, Java and Rust. 

Also see issue https://github.com/conda-forge/staged-recipes/issues/4365 and PR https://github.com/conda-forge/staged-recipes/pull/4561

### CODE REVIEW

**Comment by dirmeier** [comment]
@SoapZA

File: recipes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (bgruening) [inline recipes/meta.yaml]: I guess we should use `python -m pip install --no-deps --ignore-installed` nowadays.

Comment (dirmeier) [reply recipes/meta.yaml]: Sure I'll change that. What is the advantage of it? `pip install` goes through `setup.py install`, too and is more likely to crash if something is missing or am I wrong?

**Comment by dirmeier** [comment]
@bgruening Cool, thanks for the quick revision.

File: recipes/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/meta.yaml]: https://github.com/conda-forge/conda-forge.github.io/pull/518

**Comment by bgruening** [comment]
I might be wrong here, but as this is a pure python package I think we should do it noarch.

**Comment by SoapZA** [comment]
correct, Meson is pure python
I would maybe consider adding `ninja` as a run-time dependency, in order not to pull it in for every meson-dependent package.

**Comment by dirmeier** [comment]
@bgruening Dear Björn, can you help me out here please? Not sure how to fix this.

**Comment by bgruening** [comment]
Thanks for trying the noarch package.

**Comment by jakirkham** [comment]
In the future, please create a directory called `meson` (or whatever the package name is) and place your recipe in there. Fixing this with PR ( https://github.com/conda-forge/staged-recipes/pull/5379 ).

**Pull request merged**